airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/global_substream_cursor.py

```diff
@@ -0,0 +1,277 @@
+
+    def close_slice(self, stream_slice: StreamSlice, *args: Any) -> None:
+        """
+        Close the current stream slice.
+
+        This method is called when a stream slice is completed. For the global parent cursor, we close the child cursor
+        only after reading all slices. This ensures that we do not miss any child records from a later parent record
+        if the child cursor is earlier than a record from the first parent record.
+
+        Args:
+            stream_slice (StreamSlice): The stream slice to be closed.
+            *args (Any): Additional arguments.
+        """
+        if isinstance(stream_slice, GlobalCursorStreamSlice):
+            if stream_slice.last_slice:
+                self._lookback_window = self._timer.finish()
+                self._stream_cursor.close_slice(StreamSlice(partition={}, cursor_slice=stream_slice.cursor_slice), *args)
+        else:
+            raise ValueError(f"Stream slice must be a GlobalCursorStreamSlice, got {type(stream_slice)}")
+
+    def get_stream_state(self) -> StreamState:
+        state: dict[str, Any] = {"state": self._stream_cursor.get_stream_state()}
+
+        parent_state = self._partition_router.get_stream_state()
+        if parent_state:
+            state["parent_state"] = parent_state
+
+        if self._lookback_window is not None:
+            state["lookback_window"] = self._lookback_window
+
+        return state
+
+    def select_state(self, stream_slice: Optional[StreamSlice] = None) -> Optional[StreamState]:
+        # stream_slice is ignored as cursor is global
+        return self._stream_cursor.get_stream_state()
+
+    def get_request_params(
+        self,
+        *,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Mapping[str, Any]:
+        if stream_slice:
+            return self._partition_router.get_request_params(  # type: ignore # this always returns a mapping
+                stream_state=stream_state,
+                stream_slice=StreamSlice(partition=stream_slice.partition, cursor_slice={}),
+                next_page_token=next_page_token,
+            ) | self._stream_cursor.get_request_params(
+                stream_state=stream_state,
+                stream_slice=StreamSlice(partition={}, cursor_slice=stream_slice.cursor_slice),
+                next_page_token=next_page_token,
+            )
+        else:
+            raise ValueError("A partition needs to be provided in order to get request params")
+
+    def get_request_headers(
+        self,
+        *,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Mapping[str, Any]:
+        if stream_slice:
+            return self._partition_router.get_request_headers(  # type: ignore # this always returns a mapping
```

Comment:
Can we divide these 2 into smaller logical parts?, Like:
- get_request_headers_with_partition()
- get_request_headers_with_cursor()

Reply:
Not a blocker! But readability is bad)

Reply:
I think creating six additional methods will only worsen the readability